crates/spk-storage/src/walker.rs

```diff
@@ -0,0 +1,1155 @@
+                });
+
+                // Set up a filter function matching for the version, if any
+                if let Some(version_number) = parts.version_str {
+                    self.with_version_filter(move |version| {
+                        RepoWalkerFilter::exact_match_version_filter(
+                            version,
+                            version_number.to_string().clone(),
+                        )
+                    });
+                }
+
+                // Set up a filter function for matching the build ident, if any
+                if let Some(build_id) = parts.build_str {
+                    self.with_build_ident_filter(move |b| {
+                        RepoWalkerFilter::exact_match_build_digest_filter(
+                            b,
+                            build_id.to_string().clone(),
+                        )
+                    });
+                }
+
+                // Set up a filter function for matching the components, if any
+                if !components.is_empty() {
+                    self.with_component_filter(move |c| {
+                        RepoWalkerFilter::allowed_components_filter(c, &components)
+                    });
+                }
+            }
+            None => {
+                // Leave the filtering functions as they are.
+            }
+        }
+        Ok(self)
+    }
+
+    /// Given some file path string, this will use it to set a file
+    /// filter function that matches the file path against the start
+    /// of each file found by the walker.
+    ///
+    /// If given None instead of a path string, this will not change
+    /// the file filter function.
+    ///
+    /// This is a helper function used by the spk du command. The same
+    /// kind of filter could be setup directly by calling
+    /// with_file_filter().
+    pub fn with_file_path(&mut self, file_path: Option<String>) -> &mut Self {
+        if let Some(path) = file_path {
+            let path_pieces: Vec<String> = path.split("/").map(ToString::to_string).collect();
+            self.with_file_filter(move |f| RepoWalkerFilter::parent_paths_match(f, &path_pieces));
+        }
+        self
+    }
+
+    /// Set up a filter function for packages based on their name.
+    pub fn with_package_filter(
+        &mut self,
+        func: impl Fn(&WalkedPackage) -> bool + Send + Sync + 'a,
+    ) -> &mut Self {
+        self.package_filter_func = Arc::new(func);
+        self
+    }
+
+    /// Set up a filter function for versions based on their version
+    /// number.
+    pub fn with_version_filter(
+        &mut self,
+        func: impl Fn(&WalkedVersion) -> bool + Send + Sync + 'a,
+    ) -> &mut Self {
+        self.version_filter_func = Arc::new(func);
+        self
+    }
+
+    /// Set up a filter function for builds based on their build ident
+    /// (digest). This is separate from with_build_spec_filter()
```

Comment:
```suggestion
    /// (digest). This is separate from [`Self::with_build_spec_filter`]
```

I won't mark all of these but please do a pass